src/plugins/application_config/server/opensearch_config_client.test.ts

```diff
@@ -0,0 +1,202 @@
+      const error = new Error(ERROR_MESSAGE);
+
+      opensearchClient.asInternalUser.get.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      await expect(client.getEntityConfig('config1')).rejects.toThrowError(ERROR_MESSAGE);
+    });
+  });
+
+  describe('deleteEntityConfig', () => {
+    it('return deleted entity when opensearch deletes successfully', async () => {
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createSuccessTransportRequestPromise({});
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.deleteEntityConfig('config1');
+
+      expect(value).toBe('config1');
+    });
+
+    it('return deleted document entity when deletion fails due to index not found', async () => {
+      const error = new ResponseError({
+        statusCode: 401,
+        body: {
+          error: {
+            type: 'index_not_found_exception',
+          },
+        },
+        warnings: [],
+        headers: {
+          'WWW-Authenticate': 'content',
+        },
+        meta: {} as any,
+      });
+
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.deleteEntityConfig('config1');
+
+      expect(value).toBe('config1');
+    });
+
+    it('return deleted document entity when deletion fails due to document not found', async () => {
+      const error = new ResponseError({
+        statusCode: 401,
+        body: {
+          result: 'not_found',
+        },
+        warnings: [],
+        headers: {
+          'WWW-Authenticate': 'content',
+        },
+        meta: {} as any,
+      });
+
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.deleteEntityConfig('config1');
+
+      expect(value).toBe('config1');
+    });
+
+    it('throws error when opensearch throws error', async () => {
+      const error = new Error(ERROR_MESSAGE);
+
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      await expect(client.deleteEntityConfig('config1')).rejects.toThrowError(ERROR_MESSAGE);
+    });
+  });
+
+  describe('updateEntityConfig', () => {
```

Comment:
can we add one more test case - when entity is not present? 

Also question: Do we allow entity values such as undefined, '' empty to be passed to these APIs?

Reply:
add input validation as well as unit tests